trace/backend.go

```diff
@@ -8,6 +8,7 @@ import (
 	"time"
 
 	"github.com/golang/protobuf/proto"
+	"github.com/panjf2000/gnet/pkg/pool/byteslice"
```

Comment:
Any reason to use this over the builtin?
https://go.dev/src/sync/pool.go

Reply:
(Sorry this came from my private account) 
I can copy in what that library is doing, but I’ll answer the question of why not raw sync.Pool.

### The complexity of array pooling is the variable lengths

#### `sync.Pool`
If we just pooled slices in one pool and took one, it won’t necessarily be the size we need. If we grow it before returning it, (I think that grow causes GC of the old slice, but even if it does not,) then over time all of our arrays grow to our single largest input size.

#### `map[int]sync.Pool`

In my understanding is that the SSFSpans will be semi-random(-enough) in size, so if I default just pooling with like a key based on raw size, then we’d get a bunch of pools.

#### proposed solution

So what this library does is pool based on ceil(log2(len)).

It’s easy enough to implement from scratch, or license permitting to copy what that library has, I’d normally want to import code (.NET ecosystem) but I can adapt if we have concern about dependencies (after all, it DID bring in like 100 files for the 1 I needed).

Reply:
Given how little of the pulled in code is used, I think I should pull the code in directly (ideally I'd go contribute to the source to ship it independently, but the coordination there would be too high vs the realized cost of doing locally)